app/models/conversion_host.rb

```diff
@@ -158,14 +158,59 @@ def ipaddress(family = 'ipv4')
-    connect_ssh { |ssu| ssu.put_file(path, limits.to_json) }
+  def apply_task_limits(task_id, limits = {})
+    connect_ssh { |ssu| ssu.put_file("/var/lib/uci/#{task_id}/limits.json", limits.to_json) }
   rescue MiqException::MiqInvalidCredentialsError, MiqException::MiqSshUtilHostKeyMismatch => err
-    raise "Failed to connect and apply limits in file '#{path}' with [#{err.class}: #{err}]"
+    raise "Failed to connect and apply limits for task '#{task_id}' with [#{err.class}: #{err}]"
   rescue JSON::GeneratorError => err
     raise "Could not generate JSON from limits '#{limits}' with [#{err.class}: #{err}]"
   rescue StandardError => err
-    raise "Could not apply the limits in '#{path}' on '#{resource.name}' with [#{err.class}: #{err}]"
+    raise "Could not apply the limits for task '#{task_id}' on '#{resource.name}' with [#{err.class}: #{err}]"
+  end
+
+  # Prepare the conversion assets for a specific task.
+  #
+  # @param [Integer] id of the task that needs the preparation
+  # @param [Hash] conversion options to write on the conversion host
+  #
+  # @return [Integer] length of data written to conversion options file
+  #
+  # @raise [MiqException::MiqInvalidCredentialsError] if conversion host credentials are invalid
+  # @raise [MiqException::MiqSshUtilHostKeyMismatch] if conversion host key has changed
+  # @raise [JSON::GeneratorError] if limits hash can't be converted to JSON
+  # @raise [StandardError] if any other problem happens
+  def prepare_conversion(task_id, conversion_options)
+    filtered_options = filter_options(conversion_options)
+
+    connect_ssh do |ssu|
+      # Prepare the conversion folders
+      ssu.shell_exec("mkdir -p /var/lib/uci/#{task_id} /var/log/uci/#{task_id}", nil, nil, nil)
+
+      # Write the conversion options file
+      ssu.put_file("/var/lib/uci/#{task_id}/input.json", conversion_options.to_json)
+    end
+  rescue MiqException::MiqInvalidCredentialsError, MiqException::MiqSshUtilHostKeyMismatch => err
+    raise "Failed to connect and prepare conversion for task '#{task_id}' with [#{err.class}: #{err}]"
+  rescue JSON::GeneratorError => err
+    raise "Could not generate JSON for task '#{task_id}' from options '#{filtered_options}' with [#{err.class}: #{err}]"
+  rescue StandardError => err
+    raise "Preparation of conversion for task '#{task_id}' failed on '#{resource.name}' with [#{err.class}: #{err}]"
+  end
+
+  # Build the podman command to execute conversion
+  #
+  # @param [Integer] id of the task that conversion applies to
+  #
+  # @return [String] podman command to be executed on conversion host
+  def build_podman_command(task_id)
+    "/usr/bin/podman run --privileged"\
+    " --name conversion-#{task_id}"\
+    " --volume /dev:/dev"\
+    " --volume /var/tmp:/var/tmp"\
+    " --volume /var/lib/uci/#{task_id}:/var/lib/uci"\
+    " --volume /var/log/uci/#{task_id}:/var/log/uci"\
+    " --volume /opt/vmware-vix-disklib-distrib:/opt/vmware-vix-disklib-distrib"\
```

Comment:
Is this location guaranteed?  cc @agrare

Reply:
There is no guarantee on where vix-disklib is extracted to, you just have to symlink it to `/usr/lib`

Reply:
Oh I wonder if you can volume mount a symlink on the host :scream:

Reply:
Sorry should have been more specific, you have to symlink the .so files in that extracted directory to `/usr/lib`

E.g.:
```
$ ls -l /usr/lib/libvixDiskLib.so*
lrwxrwxrwx 1 root root 50 Oct  7 09:03 /usr/lib/libvixDiskLib.so -> /usr/lib/vmware-vix-disklib/lib64/libvixDiskLib.so
lrwxrwxrwx 1 root root 52 Oct  7 09:03 /usr/lib/libvixDiskLib.so.6 -> /usr/lib/vmware-vix-disklib/lib64/libvixDiskLib.so.6
```

Reply:
Not really in conversion host appliance. We untar it in /opt and virt-v2v-wrapper looks up for it there, so no need to care about LD_LIBRARY_PATH or to create symlinks under /usr/lib.

Reply:
:+1: